File: python/202. Happy Number/solution.py
class Solution:
    def isHappy(self, n: int) -> bool:

        def sumSquares(n):
            sum = 0

            while n != 0:
                x = n % 10
                sum += x ** 2
                n = n // 10

            return sum

        if n == 1: return True

        calculated = set()

        while n != 1 and n not in calculated:
            calculated.add(n)
            n = sumSquares(n)

        return True if n == 1 else False

            

File: java/202. Happy Number/Solution.java
class Solution {
    public boolean isHappy(int n) {
        if (n == 1) return true;
        Set<Integer> squareSums = new HashSet<>();

        while(!squareSums.contains(n)) {
            squareSums.add(n);
            n = sumSquares(n);
            if (n == 1) return true;
        }
        return false;
    }

    public int sumSquares(int number) {
        int sum = 0;
        while (number != 0) {
            int i = number % 10;
            sum +=  i * i;
            number /= 10;
        }
        return sum;
    }
}